src/actions/Scroll.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+  SelectionWithEditor,
+} from "../Types";
+import { displaySelectionDecorations } from "../editDisplayUtils";
+import { groupBy } from "../itertools";
+import { commands, TextEditor, Selection, window } from "vscode";
+import { focusEditor } from "./setSelectionsAndFocusEditor";
+
+class Scroll implements Action {
+  targetPreferences: ActionPreferences[] = [{ insideOutsideType: "inside" }];
+
+  constructor(private graph: Graph, private at: string) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    const selectionGroups = groupBy(
+      targets,
+      (t: TypedSelection) => t.selection.editor
+    );
+
+    const selections = <SelectionWithEditor[]>[];
+    const lines = <LineWithEditor[]>[];
+
+    [...selectionGroups].forEach(([editor, targets]) => {
```

Comment:
I guess you decided `runOnTargetsForEachEditor` was overkill because of the promise stuff?

Also,
```suggestion
    selectionGroups.forEach(([editor, targets]) => {
```

Reply:
Is just because I don't have a typed selection. I just have the line and the editor and I didn't want to actually recreate entire context